kymatio/scattering1d/core/scattering1d.py

```diff
@@ -28,6 +29,9 @@ def scattering1d(x, pad, unpad, backend, J, psi1, psi2, phi, pad_left=0,
         The array `phi[j]` is a real-valued filter.
     J : int
         scale of the scattering
+    log2_T : int
```

Comment:
`T` can be non-dyadic. `log2_T` is also reused so eases readability / searchability.

Reply:
so if T can be non-dyadic, why specify it as its log?

Reply:
Because subsampling, and sampling filters, is only done on dyadic scales. The "non-dyadic" part is for setting `sigma_low`, i.e. the amount of imposed time-shift invariance does not have to be in powers of 2.

It also sets up for `log2_F` in JTFS which is reused a _lot_ more and serves more diverse logic.